### DISCUSSION TITLE
Pico Sim868

### DISCUSSION THREAD

**Discussion by Manjo80** [micropython/micropython#10385] [Hardware & Peripherals]
Hallo, I new in python / micropython … To lean it, my actual project is creating an sms buttler.
I‘m using an Raspberry Pico board. For the moment the only thing i have is am able to switch my mosfets by buttons.
Now I wanted to start with the pico sim868 modul.
I tryed on different way but nothing worked at the moment.
For the moment I‘m trying it with the serial library, but I can‘t find a library, where everything is working. On using the micropython-serial library I‘m stucked with the termios library, what seems to be a dummy library, but the command are not working without in the serial.py and on using the pyserial library im stucked with the ffi.py where I‘m receiving an error.
I‘have also tryed over uart with the machine library but like this I wasn‘t also able to connect to the sim868, an send an test sms (to see if it works).
I here someone how could give me a tips to solve my problem to be able to go on with my little project

**Comment by Manjo80** [comment]
Now I‘using the sim800 lib and I am able to connect to the module.
Sending at command is working but I have a other problem I think
I‘m using this skript to check the module

```ruby
import sim800

# Initialize the SIM800 module with UART pin 0 and BAUD rate 115200
gsm = sim800.sim800(0, 115200)

# Check network registration
reg = gsm.gsm.write("AT+CREG?\r")
print("Network registration:", reg)

# Check operator selection
ops = gsm.gsm.write("AT+COPS?\r")
print("Operator selection:", ops)

# Check GPRS attachment
att = gsm.gsm.write("AT+CGATT?\r")
print("GPRS attachment:", att)

# Check local time synchronization
lts = gsm.gsm.write("AT+CLTS?\r")
print("Local time synchronization:", lts)

# Check mobile equipment event reporting
mer = gsm.gsm.write("AT+CMER?\r")
print("Mobile equipment event reporting:", mer)

# Check signal quality
csq = gsm.gsm.write("AT+CSQ\r")
print("Signal quality:", csq)

```


The ansers are


Network registration: 9
Operator selection: 9
GPRS attachment: 10
Local time synchronization: 9
Mobile equipment event reporting: 9
Signal quality: 7


I have checked the sim card an tryed even with an other sim card. Everytime the same result.

Also with the demo script form waveshare I can‘t register to the network



And in the sim800 lib I had 3 errors

Line 27:

```ruby

def delete_message(self, index) 

```
with missing : at the end.


Line 31: 

```ruby
   Def delete_all_messages(self)

```

the same
Line 35: 

```ruby

self.gsm.write('AT+CMGS="{}"') 

```

needed ro be changed to should 

```ruby
self.gsm.write('AT+CMGS="{}"'.format(number)) 
```

**Reply by robert-hh** [reply]
For proper formatting please enclose code in lines with three backticks, or indent the whole code block.
